lib/data_integrity_utils.pm

```diff
@@ -19,22 +19,43 @@ use testapi;
 use File::Basename;
 use Digest::file 'digest_file_hex';
 
-our @EXPORT = 'verify_checksum';
+our @EXPORT = qw(verify_checksum get_image_digest);
+
+=head2
+Returns image digest. Digest retrieval is platform-specific depends.
+=cut
+sub get_image_digest {
+    my ($image_path) = shift;
+
+    my $digest;
+    if (check_var('BACKEND', 'svirt')) {
+        $digest = console('svirt')->get_cmd_output("sha256sum $image_path");
+        # On Hyper-V the hash starts with '\'
+        my $start = check_var('VIRSH_VMM_FAMILY', 'hyperv') ? 1 : 0;
+        $digest = substr $digest, $start, 64;    # extract SHA256 from the output
```

Comment:
you can return the $digest here (as it is better in Perl not declare variables if it is not really needed, for maintenance)

Reply:
I find it a matter of taste. I'll keep it like this. Thanks.